app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,44 @@
+class BookingsController < ApplicationController
+  before_action :set_event
+  before_action :set_booking, only: %i[update cancelation]
+
+  def create
+    @booking = @event.bookings.new(booking_params)
+
+    if @booking.save
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @booking.update(booking_params)
+      redirect_to root_path
+    else
+      render "pages/home", status: :unprocessable_entity
+    end
+  end
```

Comment:
canceled? is the only attribute that can be updated and you already created the cancel action so we have no need for an update action. You can remove it

Reply:
deleted